cmds/modules/api_gateway/main.go

```diff
@@ -128,6 +137,85 @@ func action(cli *cli.Context) error {
 		Msg("starting api-gateway module")
 
 	// block forever
-	<-ctx.Done()
-	return nil
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+			// check if we need to run an update on the peer and only do the update if all the changes are done successfully
+		case <-time.After(10 * time.Minute):
+
+			log.Debug().Msg("checking if node is maintaining a healty substrate connection")
+			cl, _, err := manager.Raw()
+			if err == nil {
+				// skip update if the connection is working properly
+				log.Debug().Msg("the open connection is healty, no update needed")
+				cl.Client.Close()
+				continue
+			}
+
+			env, err := environment.Get()
+			if err != nil {
+				// skip update if we can't get env
+				log.Debug().Err(err).Msg("failed to load node environment")
+				continue
+			}
+
+			updatedSubURLs := env.SubstrateURL
+			updatedRelayURLs := environment.GetRelaysURLs()
+
+			// make sure urls are sorted for comparison
+			slices.Sort(subURLs)
+			slices.Sort(relayURLs)
+			slices.Sort(updatedSubURLs)
+			slices.Sort(updatedRelayURLs)
+
+			// skip update if substrate and relay urls did not change
+			if slices.Equal(subURLs, updatedSubURLs) && slices.Equal(relayURLs, updatedRelayURLs) {
+				log.Debug().Msg("zos-config doesn't have updated config to update the node with")
+				continue
+			}
+
+			log.Debug().Strs("relays_urls", updatedRelayURLs).Strs("substrate_urls", updatedSubURLs).Msg("detected new update in configuration")
+			newPeerCtx, newCancel := context.WithCancel(ctx)
+
+			newManager, err := environment.GetSubstrate()
+			if err != nil {
+				// skip update if can't get sub manager
+				log.Debug().Err(err).Msg("failed to get substrate manager")
+				continue
+			}
+
+			if _, err = peer.NewPeer(
+				newPeerCtx,
+				hex.EncodeToString(pair.Seed()),
+				newManager,
+				router.Serve,
+				peer.WithKeyType(peer.KeyTypeEd25519),
+				peer.WithRelay(updatedRelayURLs...),
+				peer.WithInMemoryExpiration(6*60*60), // 6 hours
+			); err != nil {
+				log.Debug().Err(err).Msg("failed to start a new rmb peer")
+				continue
+			}
+
+			if !slices.Equal(subURLs, updatedSubURLs) {
```

Comment:
newPeer already created with the newManger, so we need to update the gw to use the newManger anyway?